schemas/pl72_run_start.fbs

```diff
@@ -0,0 +1,33 @@
+// Run start message
+//
+// Typical producers and consumers:
+// Produced by NICOS and NeXus-Streamer
+// Consumed by NeXus file writer - to trigger start of file writing
+// Consumed by Mantid - marks the start of a run, Mantid has various options for
+//                      runs, such as stopping listening at the end of the current
+//                      run or populating a new workspace for each new run
+
+include "df12_det_spec_map.fbs";
+
+file_identifier "pl72";
+
+table RunStart {                                     //  *Mantid*    // *File Writer* // *Description*
+    start_time : uint64;                             //  Required    //  Optional     // nanoseconds since Unix epoch (1 Jan 1970)
+    stop_time : uint64;                              //  Unused      //  Optional     // nanoseconds since Unix epoch (1 Jan 1970), optional, can send a RunStop instead
+    run_name : string;                               //  Required    //  Unused       // Name for the run, used as workspace name by Mantid
+    instrument_name : string;                        //  Required    //  Unused       // Name of the instrument, only required by Mantid
+    nexus_structure : string;                        //  Optional    //  Required     // JSON description of NeXus file (See https://github.com/ess-dmsc/kafka-to-nexus/ for detail)
```

Comment:
Getting instrument name from the nexus structure would cause us a headache at ISIS for the time being. We aren't able to provide nexus_structure for every instrument yet so not being able to fall back on IDFs would hold us back from at least getting things mostly working on many instruments. We could send a nexus stucture with just the instrument name, but then we'd need to detect that we have only that and not a full geometry description somehow. It would be a lot of complexity introduced for little gain.

job_id must be unique, run_name would be a run number at ISIS which is, at best, unique only per instrument. I'm finding out if instrument_name + run_name would work, or whether the user can reuse run_name for rejected runs etc.

service_id would not be needed if we move to the pool idea, but the field can be marked `deprecated` at that point. I don't think we can drop it right now.

Reply:
I've checked with Freddie and `instrument_name + run_name` is used as a unique identifier for files at ISIS, however it would not work as a unique write job ID because it gets reused if a run is aborted and restarted. I can't see any way to drop any of these fields without net complexity increasing. We could discuss at a meeting though.